WMF Framework/AlertType.swift

```diff
@@ -0,0 +1,21 @@
+public enum AlertType {
+    case listLimitExceeded(limit: Int)
+    case entryLimitExceeded(limit: Int)
+    case genericNotSynced
+    case downloading
+}
+
+extension AlertType: Equatable {
+    public static func ==(lhs: AlertType, rhs: AlertType) -> Bool {
+        switch (lhs, rhs) {
+        case let (.listLimitExceeded(a), .listLimitExceeded(b)),
+             let (.entryLimitExceeded(a), .entryLimitExceeded(b)):
+            return a == b
```

Comment:
For equality do we care if a == b or is it enough to know both are of same type?

Reply:
We have to compare associated values, it's not enough to compare types

Reply:
@montehurd feel free to post a solution without Equatable 👍

Reply:
You are right on 👍 about equatable and associated values: 
https://andybargh.com/swift-enumerations/#Boolean_Comparison_of_Associated_Values_Enumerations

Given that it sounded like we want this equality check to be associated value agnostic, it looks like it can be simplify it to this:

     extension ReadingListAlertType: Equatable {
          public static func ==(lhs: ReadingListAlertType, rhs: ReadingListAlertType) -> Bool {
               switch (lhs, rhs) {
                    case  (.listLimitExceeded, .listLimitExceeded),
                          (.entryLimitExceeded, .entryLimitExceeded),
                          (.genericNotSynced, .genericNotSynced),
                          (.downloading, .downloading):
                         return true
                    default:
                         return false
               }
          }
     }

Being able to do multi-argument switches still feels like magic :)